enos/modules/solo_worker/main.tf

```diff
@@ -0,0 +1,178 @@
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+locals {
+  tags = merge(
+    var.common_tags,
+    {
+      "PetName" = random_pet.worker.id
+    },
+  )
+  selected_az = data.aws_availability_zones.available.names[random_integer.az.result]
+}
+
+resource "random_string" "cluster_id" {
+  length  = 8
+  upper   = false
+  numeric = false
+  special = false
+}
+
+resource "random_pet" "worker" {
+  separator = "_"
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+  filter {
+    name   = "zone-name"
+    values = var.availability_zones
+  }
+}
+
+data "aws_availability_zone" "worker_az" {
+  name = local.selected_az
+}
+
+data "aws_kms_key" "kms_key" {
+  key_id = var.kms_key_arn
+}
+
+resource "random_integer" "az" {
+  min = 0
+  max = length(data.aws_availability_zones.available.names) - 1
+  keepers = {
+    # Generate a new integer each time the list of aws_availability_zones changes
+    # keepers have to be strings, sort the list in case order changes but zones don't
+    listener_arn = join("", sort(data.aws_availability_zones.available.names))
+  }
+}
+
+resource "aws_subnet" "default" {
```

Comment:
Maybe a comment here to clarify that this is a separate subnet from the created ones in infra